docs/contrib/PlatformBringUp.md

```diff
@@ -0,0 +1,415 @@
+
+Follow the steps below to create the necessary directories and files. **NOTE: riscv32i is just an example and not a required name.**:
+
+
+```
+$ mkdir -p OpenROAD-flow-scripts/flow/designs/MyNewPlatform/riscv32i
+$ touch OpenROAD-flow-scripts/flow/designs/MyNewPlatform/riscv32i/config.mk
+$ touch OpenROAD-flow-scripts/flow/designs/MyNewPlatform/riscv32i/constraints.sdc
+```
+
+
+This creates two directories MyNewPlatform and riscv32i and two empty files `config.mk` and `constraints.sdc` in OpenROAD-flow-scripts/flow/designs/MyNewPlatform/riscv32i. 
+
+
+## Platform Configuration
+
+This section describes the necessary files in the platform directory needed for the OpenROAD flow. Specifically the `config.mk` file in the platform directory has all of the configuration variables that the flow uses. Refer to the OpenROAD-flow-scripts documentation for a full list of configuration variables that can be set.
+
+For an example of a platform config.mk file, refer to `OpenROAD-flow-scripts/flow/platforms/sky130hd/config.mk.`
+
+
+## Design Configuration
+
+This section describes files in the design directory.
+
+
+## config.mk
+
+The `config.mk` file describes design-specific variables.
+
+For Example:
+
+
+```
+    DESIGN_NAME
+    PLATFORM
+    VERILOG_FILES
+    SDC_FILE
+    CORE_UTILIZATION
+    CORE_ASPECT_RATIO
+    CORE_MARGIN
+    PLACE_DENSITY
+```
+
+Alternatively, in place of `CORE_UTILIZATION` and `CORE_ASPECT_RATIO`, `DIE_AREA` and `CORE_AREA` can be specified instead.
+
+
+Following is a sample `config.mk` file for the `riscv32i` design:
+
+
+```
+#config.mk
+###########################
+export DESIGN_NAME     = riscv
+export PLATFORM        = sky130hd
+
+export VERILOG_FILES = $(sort $(wildcard ./designs/src/$(DESIGN_NICKNAME)/*.v))
+
+export SDC_FILE = ./designs/$(PLATFORM)/$(DESIGN_NICKNAME)/constraint.sdc
+
+export CORE_UTILIZATION  = 30
+export CORE_ASPECT_RATIO = 1
+export CORE_MARGIN       = 2
+export PLACE_DENSITY     = 0.70
+```
+
+
+Refer to the [OpenSTA][https://github.com/The-OpenROAD-Project/OpenSTA/blob/master/doc/OpenSTA.pdf] documenation to change these variables to suit the needs of the specific design being made.
+
+
+## constraint.sdc
+
+The `constraint.sdc` file defines timing constraints for the design. Here’s an example of a constraint.sdc file which defines a clock `clk` with a period of 8.4 nanoseconds.
+
+
+```
+#constraint.sdc
+############################
+create_clock [get_ports clk] -period 8.4
+```
+
+
+The `create_clock` command allows you to define clocks that are either connected to nets  or are virtual and can be customized.
+
+
+```
+#Create a netlist clock
+create_clock -period <float> <netlist clock list of regexes>
+
+#Create a virtual clock
+create_clock -period <float> -name <virtual clock name>
+
+#Create a netlist clock with custom waveform/duty-cycle
+create_clock -period <float> -waveform {rising_edge falling_edge} <netlist clock list of regexes>
```

Comment:
I don't think cts currently supports anything other than a 50/50 duty cycle. This command would create the clock constraints, but cts would probably discard the waveform. @maliberty?

Reply:
Even though this feature might not be implemented yet should we keep the example for the future when it is implemented?